internal/resource/config/oauth/client/oauth_client_resource.go

```diff
@@ -0,0 +1,1579 @@
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func (r *oauthClientResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	var model oauthClientResourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &model)...)
+
+	// Persistent Grant Expiration Validation
+	if internaltypes.IsDefined(model.PersistentGrantExpirationType) && model.PersistentGrantExpirationType.ValueString() != "OVERRIDE_SERVER_DEFAULT" {
+		if !internaltypes.IsDefined(model.PersistentGrantExpirationTime) || !internaltypes.IsDefined(model.PersistentGrantExpirationTimeUnit) {
+			resp.Diagnostics.AddError("persistent_grant_expiration_type must be configured to \"OVERRIDE_SERVER_DEFAULT\" to modify the other persistent_grant_expiration values.", "")
+		}
+	}
+	if (internaltypes.IsDefined(model.PersistentGrantExpirationTime) || internaltypes.IsDefined(model.PersistentGrantExpirationTimeUnit)) && model.PersistentGrantExpirationType.ValueString() != "OVERRIDE_SERVER_DEFAULT" {
+		resp.Diagnostics.AddError("persistent_grant_expiration_type must be configured to \"OVERRIDE_SERVER_DEFAULT\" to modify the other persistent_grant_expiration values.", "")
+	}
+
+	// Refresh Token Rolling Validation
+	if model.RefreshTokenRollingIntervalType.ValueString() == "OVERRIDE_SERVER_DEFAULT" && !internaltypes.IsDefined(model.RefreshTokenRollingInterval) {
+		resp.Diagnostics.AddError("refresh_token_rolling_interval must be configured when refresh_token_rolling_interval_type is \"OVERRIDE_SERVER_DEFAULT\".", "")
+	}
+	if internaltypes.IsDefined(model.RefreshTokenRollingInterval) && model.RefreshTokenRollingIntervalType.ValueString() != "OVERRIDE_SERVER_DEFAULT" {
+		resp.Diagnostics.AddError("refresh_token_rolling_interval_type must be configured to \"OVERRIDE_SERVER_DEFAULT\" to modify the refresh_token_rolling value.", "")
+	}
+
+	//  Client Auth Defined
+	var clientAuthDefined bool
+	var clientAuthAttributes map[string]attr.Value
+	if internaltypes.IsDefined(model.ClientAuth) {
+		clientAuthDefined = true
+		clientAuthAttributes = model.ClientAuth.Attributes()
+		if internaltypes.IsDefined(clientAuthAttributes["type"]) {
+			clientAuthType := clientAuthAttributes["type"].(types.String).ValueString()
+			switch clientAuthType {
+			case "PRIVATE_KEY_JWT":
+				if !internaltypes.IsNonEmptyObj(model.JwksSettings) {
+					resp.Diagnostics.AddError("jwks_settings must be defined when client_auth is configured to \"PRIVATE_KEY_JWT\".", "")
+				}
+			case "CERTIFICATE":
+				if !internaltypes.IsDefined(clientAuthAttributes["client_cert_subject_dn"]) || !internaltypes.IsDefined(clientAuthAttributes["client_cert_issuer_dn"]) {
+					resp.Diagnostics.AddError("client_cert_subject_dn and client_cert_issuer_dn must be defined when client_auth is configured to \"CERTIFICATE\".", "")
+				}
+			}
+		}
+	} else {
+		clientAuthDefined = false
```

Comment:
You can set `clientAuthDefined = internaltypes.IsDefined(model.ClientAuth)` rather than using an if-else